libs/api/domains/driving-license/src/lib/drivingLicense.service.ts

```diff
@@ -10,153 +10,207 @@ import {
-      id: type.nr,
-      name: type.heiti,
-    }))
+  async getRemarkTypes(): Promise<RemarkType[]> {
+    const types = await this.drivingLicenseApi.apiOkuskirteiniTegundirathugasemdaGet(
+      {},
+    )
+
+    return types.map(
+      (type: TegundAthugasemdaDto) =>
+        ({
+          id: parseInt(type?.nr as string, 10),
+          remark: type.athugasemd,
+          name: type.heiti,
+          description: type.lysing,
+          // TODO: I have no idea what this even is
+          for: type.giltFyrir,
+        } as RemarkType),
+    )
   }
 
   async getPenaltyPointStatus(
     nationalId: User['nationalId'],
   ): Promise<PenaltyPointStatus> {
-    const status = await this.drivingLicenseApi.getPenaltyPointStatus(
-      nationalId,
+    const status = await this.drivingLicenseApi.apiOkuskirteiniPunktastadaKennitalaGet(
+      {
+        kennitala: nationalId,
+      },
     )
+
     return {
       nationalId,
-      isPenaltyPointsOk: status.iLagi,
+      // TODO: fix
+      // return type is wrong in open api doc
+      isPenaltyPointsOk: status.iLagi as boolean,
     }
   }
 
   async getTeachingRights(
     nationalId: User['nationalId'],
   ): Promise<TeachingRightsStatus> {
-    const status = await this.drivingLicenseApi.getTeachingRights(nationalId)
+    const status = await this.drivingLicenseApi.apiOkuskirteiniHasteachingrightsKennitalaGet(
+      {
+        kennitala: nationalId,
+      },
+    )
 
     return {
       nationalId,
-      hasTeachingRights: status.value > 0,
+      hasTeachingRights: status > 0,
     }
   }
 
   async getListOfJuristictions(): Promise<Juristiction[]> {
-    const embaetti = await this.drivingLicenseApi.getListOfJuristictions()
+    const embaetti = await this.drivingLicenseApi.apiOkuskirteiniEmbaettiGet({})
 
-    return embaetti.map(({ nr, postnumer, nafn }) => ({
-      id: nr,
-      zip: postnumer,
-      name: nafn,
+    return embaetti.map(({ nr, postnumer, nafn }: EmbaettiDto) => ({
+      id: nr || 0,
+      zip: postnumer || 0,
+      name: nafn || '',
     }))
   }
 
+  private async getDrivingAssessmentResult(
+    nationalId: string,
+  ): Promise<AkstursmatDto | null> {
+    try {
+      return await this.drivingLicenseApi.apiOkuskirteiniSaekjaakstursmatKennitalaGet(
+        {
+          kennitala: nationalId,
+        },
+      )
+    } catch (e) {
+      if (e.status === 404) {
+        return null
+      }
+
+      throw e
+    }
+  }
+
   async getApplicationEligibility(
     nationalId: string,
     type: DrivingLicenseType['id'],
   ): Promise<ApplicationEligibility> {
-    const assessmentResult = await this.drivingLicenseApi.getDrivingAssessment(
-      nationalId,
-    )
-    const hasFinishedSchoolResult = await this.drivingLicenseApi.hasFinishedSchool(
-      nationalId,
+    const assessmentResult = await this.getDrivingAssessmentResult(nationalId)
+
+    const hasFinishedSchoolResult = await this.drivingLicenseApi.apiOkuskirteiniKennitalaFinishedokugerdiGet(
```

Comment:
is there a capital O missing in the end of the function name?
FinishedOkugerdi ?
> Finishedokugerdi

Reply:
this is a generated client from the api docs.. if it's like this in the webservice, it's lice that in the function name